lxd/container_lxc.go

```diff
@@ -1402,24 +1486,54 @@ func (c *containerLXC) expandConfig() error {
 func (c *containerLXC) expandDevices() error {
 	devices := types.Devices{}
 
+	var storagePoolsInProfile []string
 	// Apply all the profiles
 	for _, p := range c.profiles {
 		profileDevices, err := dbDevices(c.daemon.db, p, true)
 		if err != nil {
 			return err
 		}
 
+		// COMMENT(brauner): Check all pools specified in the
+		// container's profiles.
 		for k, v := range profileDevices {
+			if v["type"] == "disk" && v["path"] == "/" {
+				if v["pool"] != "" && !shared.StringInSlice(v["pool"], storagePoolsInProfile) {
+					storagePoolsInProfile = append(storagePoolsInProfile, v["pool"])
+				}
+			}
 			devices[k] = v
 		}
 	}
 
+	// COMMENT(brauner): Check if only one storage pool, the container is
+	// supposed to be created on is specified in its profiles.
+	if len(storagePoolsInProfile) > 1 {
+		return fmt.Errorf("Profiles the container belongs to specify different storage pools.")
+	}
+
+	storagePoolInLocalDev := ""
 	// Stick local devices on top
 	for k, v := range c.localDevices {
+		if v["type"] == "disk" && v["path"] == "/" {
+			storagePoolInLocalDev = v["pool"]
+		}
 		devices[k] = v
 	}
 
+	if len(storagePoolsInProfile) == 1 {
+		if storagePoolsInProfile[0] != "" && storagePoolInLocalDev != "" {
+			if storagePoolsInProfile[0] != storagePoolInLocalDev {
+				return fmt.Errorf("Conflicting storage pool specified in local and profile devices.")
+			}
+		}
+		c.storagePool = storagePoolsInProfile[0]
+	} else {
+		c.storagePool = storagePoolInLocalDev
+	}
```

Comment:
There can be a root device in the local devices and a root device in the profiles. So the user might want a container to be attached to a specific profile that specifies a root device but he might also have given an explicit pool the container is supposed to live on. We need to deal with that case. Container's cannot be attached to a different pool then the profile they live on. (Again, this is why I think this is fragile.)